casanovo/denovo/dataloaders.py

```
@@ -342,6 +355,288 @@ def _make_dataset(
+                                scan_window_array, dtype=np.float32
+                            ),
+                            "ms_array": np.asarray(ms_array, dtype=np.int8),
+                        }
+
+                        if not annotated:
+                            record["charge"] = charge
+
+                        yield pl.DataFrame(record)
+
+            logging.warning(
+                f"{skipped} spectra were skipped due to missing MS1 scans for file {spectra}"
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)